opam-ci-check/lib/lint.ml

```diff
@@ -367,40 +362,51 @@ module Checks = struct
     in
     if newly_published then checks @ newly_published_checks else checks
 
-  let run_checks ~repo_dir ~pkg ~packages ?(newly_published = false) opam =
-    checks ~newly_published packages ~repo_dir
+  let lint_package ~opam_repo_dir ~pkg ~pkg_src_dir ~repo_package_names
+      ~newly_published opam =
+    checks ~newly_published ~opam_repo_dir ~pkg_src_dir repo_package_names
     |> List.map (fun f -> f ~pkg opam)
     |> List.concat
-
-  let parse_error pkg = (pkg, ParseError)
 end
 
-let get_packages repo_dir =
+type t = {
+  pkg : OpamPackage.t;
+  newly_published : bool option;
+  pkg_src_dir : string option;
+  opam : OpamFile.OPAM.t;
+}
+
+let v ~pkg ?(newly_published = None) ~pkg_src_dir opam =
+  { pkg; newly_published; pkg_src_dir; opam }
+
+(** A package is considered newly published if the repository doesn't have any
+    other versions of the package, other than the current one.
+
+    NOTE: If two versions of a package are published in the same PR, this
+    inference would fail to detect the package as new.*)
```

Comment:
This is a very helpful pointer! :+1:

Reply:
It's possible to improve the inference in the cases where all the changed versions of a package have been passed to `lint`, but I'm not sure if it's worth fixing things for that edge case, right now.